gnd/src/main/java/com/google/android/gnd/ui/map/gms/GoogleMapsMapAdapter.java

```diff
@@ -470,25 +470,18 @@ private int parseColor(@Nullable String colorHexCode) {
   }
 
   private void onCameraIdle() {
-    cameraTargetBeforeDrag = null;
-  }
-
-  private void onCameraMoveStarted(int reason) {
-    if (reason == REASON_DEVELOPER_ANIMATION) {
-      // MapAdapter was panned by the app, not the user.
-      return;
+    if (cameraChangeReason == REASON_GESTURE) {
+      LatLng target = map.getCameraPosition().target;
+      float zoom = map.getCameraPosition().zoom;
+      cameraMoves.onNext(new CameraPosition(fromLatLng(target), zoom));
+      cameraChangeReason = REASON_DEVELOPER_ANIMATION;
     }
-    cameraTargetBeforeDrag = map.getCameraPosition().target;
   }
 
-  private void onCameraMove() {
-    com.google.android.gms.maps.model.CameraPosition gmsCameraPosition = map.getCameraPosition();
-    Point target = fromLatLng(gmsCameraPosition.target);
-    CameraPosition position = new CameraPosition(target, gmsCameraPosition.zoom);
-    cameraMoves.onNext(position);
-    if (cameraTargetBeforeDrag != null
-        && !gmsCameraPosition.target.equals(cameraTargetBeforeDrag)) {
-      dragInteractions.onNext(target);
+  private void onCameraMoveStarted(int reason) {
+    cameraChangeReason = reason;
+    if (reason == REASON_GESTURE) {
+      dragInteractions.onNext(true);
```

Comment:
If the value isn't meaningful consider using `NIL` in our custom `Nil` class instead.